src/Components/Members/Create/index.jsx

```diff
@@ -0,0 +1,234 @@
+import React from 'react';
+import styles from './addMembers.module.css';
+import { Input } from '../../Shared/Inputs';
+import { useState } from 'react';
+import { Link } from 'react-router-dom';
+import Button from '../../Shared/Button';
+import Modal from '../../Shared/Modal';
+
+const MembersCreate = () => {
+  const [firstName, setFirstName] = useState('');
+  const [lastName, setLastName] = useState('');
+  const [dni, setDni] = useState('');
+  const [phone, setPhone] = useState('');
+  const [email, setEmail] = useState('');
+  const [city, setCity] = useState('');
+  const [birthDay, setBirthday] = useState('');
+  const [postalCode, setPostalCode] = useState('');
+  const [isActive, setIsActive] = useState(true);
+  const [membership, setMembership] = useState('Classic');
+
+  const [message, setMessage] = useState('');
+  const [showModal, setShowModal] = useState(false);
+
+  const handleFirstName = (e) => {
+    setFirstName(e.target.value);
+  };
+
+  const handleLastName = (e) => {
+    setLastName(e.target.value);
+  };
+
+  const handleDniChange = (e) => {
+    const value = e.target.value;
+    setDni(value);
+  };
+
+  const handlePhoneChange = (e) => {
+    const value = e.target.value;
+    setPhone(value);
+  };
+
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+  };
+
+  const handleCityChange = (e) => {
+    setCity(e.target.value);
+  };
+
+  const handleBirthdayChange = (e) => {
+    setBirthday(e.target.value);
+  };
+
+  const handlePostalCodeChange = (e) => {
+    setPostalCode(e.target.value);
+  };
+
+  const handleMembershipChange = (e) => {
+    setMembership(e.target.value);
+  };
+
+  const addMember = async ({
+    firstName,
+    lastName,
+    dni,
+    phone,
+    email,
+    city,
+    birthDay,
+    postalCode,
+    isActive,
+    membership
+  }) => {
+    try {
+      const newMembers = await fetch(`${process.env.REACT_APP_API_URL}/api/members/`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          firstName,
+          lastName,
+          dni,
+          phone,
+          email,
+          city,
+          birthDay,
+          postalCode,
+          isActive,
+          membership
+        })
+      });
+
+      const data = await newMembers.json();
+      setMessage(data.message);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    addMember({
+      firstName,
+      lastName,
+      dni,
+      phone,
+      email,
+      city,
+      birthDay,
+      postalCode,
+      isActive,
+      membership
+    });
+    setFirstName('');
+    setLastName('');
+    setDni('');
+    setPhone('');
+    setEmail('');
+    setCity('');
+    setBirthday('');
+    setPostalCode('');
+    setIsActive(true);
+    setMembership('Classic');
+    setShowModal(true);
```

Comment:
be careful, if you make a mistake while filling out the form, it will empty all the fields and require you to fill them out again.

Reply:
Moved to the addMember function in case of success only